pandas/core/categorical.py

```diff
@@ -608,21 +578,17 @@ def _set_categories(self, categories, fastpath=False):
 
         """
 
-        categories = self._validate_categories(categories, fastpath=fastpath)
-        if (not fastpath and self._categories is not None and
-                len(categories) != len(self._categories)):
+        if fastpath:
```

Comment:
lets deprecate ``fastpath`` (and convert fastpath -> dtype construction, and instead allow ``dtype`` to be passed